### PR TITLE
fixes #1766

### LINKED ISSUE #1766: Refactor rust-buffered-queue.h
This is a util class and should be moved into the util folder.

### PR DESCRIPTION
**Pull request by SainiAditya1**
Thank you for making Rust GCC better!

If your PR fixes an issue, you can add "Fixes #issue_number" into this
PR description and the git commit message. This way the issue will be
automatically closed when your PR is merged. If your change addresses
an issue but does not fully fix it please mark it as "Addresses #issue_number"
in the git commit message.

Here is a checklist to help you with your PR.

- \[x] GCC development requires copyright assignment or the Developer's Certificate of Origin sign-off, see https://gcc.gnu.org/contribute.html or https://gcc.gnu.org/dco.html
- \[x] Read contributing guidlines
- \[x] `make check-rust` passes locally
- \[x] Run `clang-format`
- \[x] Added any relevant test cases to `gcc/testsuite/rust/`

Note that you can skip the above if you are just opening a WIP PR in
order to get feedback.
---

*Please write a comment explaining your change. This is the message
that will be part of the merge commit.

### CODE REVIEW

File: gcc/rust/util/rust-buffered-queue.h [hunk 1]
```diff
@@ -72,7 +72,6 @@ template <typename T, typename Source> class buffered_queue
 		  end = num_queued_items;
 		  // TODO: would move be better here? optimisation for move with
 		  // shared pointer?
-
```
Comment (philberty) [inline gcc/rust/util/rust-buffered-queue.h:75]: can you remove this empty line deletion as it separates the TODO comment with the comment on the next line

**Review by philberty — CHANGES_REQUESTED** [review]
LGTM though i would like that line deletion to be removed.

Also you need to update your opening PR comment here to remove the check list otherwise this will be added into the bors merge commit.

**Comment by philberty** [comment]
Also I see that you have done this in two commits this should be squashed into 1 commit. You can do this via:

```git rebase -i HEAD~2```

File: gcc/rust/util/rust-buffered-queue.h (hunk 1 above)
Comment (SainiAditya1) [reply gcc/rust/util/rust-buffered-queue.h:75]: @philberty , ok I will fix it.

**Comment by philberty** [comment]
@SainiAditya1 I think this pr should be a:

```git mv rust-buffered-queue.h util/```

There shouldn't be any edits to the file it self

**Pull request closed without merging**